cmflib/cmf_merger.py

```diff
@@ -17,103 +17,141 @@
+                    # Handle the case where the context already exists, possibly due to concurrent pushes.
+                    # As both pipelines will be unable to fetch data from server 
+                    # updating custom properties if context already exists
+                    _ = cmf_class.update_context(
+                        str(stage["type"]),
+                        str(stage["name"]),
+                        stage["id"],
+                        stage["properties"],
+                        custom_properties=stage["custom_properties"]
                     )
-                elif artifact_type == "Model" and event_type == 3:
-                    props["uri"] = uri
-                    cmf_class.log_model_with_version(
-                        path=artifact_name,
-                        event="input",
-                        props=props,
-                        custom_properties=custom_props,
+                except Exception as e:
+                    print(f"Error in merge_created_context")
+                try:
+                    _ = cmf_class.merge_created_execution(
+                        execution["properties"]["Context_Type"],
+                        execution["properties"]["Execution"],
+                        execution["properties"],
+                        execution["custom_properties"],
+                        execution["name"]
                     )
-                elif artifact_type == "Model" and event_type == 4:
-                    props["uri"] = uri
-                    cmf_class.log_model_with_version(
-                        path=artifact_name,
-                        event="output",
-                        props=props,
-                        custom_properties=custom_props,
+                except AlreadyExistsError as e:
+                    _ = cmf_class.update_execution(
+                        execution["id"],
+                        execution["custom_properties"] 
                     )
-                elif artifact_type == "Metrics":
-                    # print(props,'parse')
-                    cmf_class.log_execution_metrics_from_client(event["artifact"]["name"], custom_props)
-                elif artifact_type == "Dataslice":
-                    dataslice = cmf_class.create_dataslice(event["artifact"]["name"])
-                    dataslice.commit_existing(uri, props, custom_props)
-                elif artifact_type == "Step_Metrics":
-                    cmf_class.commit_existing_metrics(event["artifact"]["name"], uri, props, custom_props)
-                else:
-                    pass
+                except Exception as e:
+                    print(f"Error in merge_created_execution {e}")
+
+                for event in execution["events"]:  # Iterates over all the events
+                    artifact_type = event["artifact"]["type"]
+                    event_type = event["type"]
+                    artifact_name = (event["artifact"]["name"].split(":"))[0]
```

Comment:
Put a helper function get_artifact_name and put the split logic there with proper comments.  wherever we do the split , we should call the common function instead doing it seperately

Reply:
As I checked, split function is used only once in the file and for split functions in other files will be creating helper functions in next PR. Have added to our tasks sheet.